src/library/serato/seratofeature.cpp

```diff
@@ -0,0 +1,1065 @@
+    //
+    // drive.canonicalPath() make a system call to the underlying filesystem
+    // introducing delay if it is unreadable.
+    // drive.filePath() doesn't make any access to the filesystem and consequently
+    // shorten the delay
+    databaseLocations.append(QDir::drives());
+#elif defined(__LINUX__)
+    // To get devices on Linux, we look for directories under /media and
+    // /run/media/$USER.
+
+    // Add folders under /media to devices.
+    databaseLocations += QDir(QStringLiteral("/media")).entryInfoList(QDir::AllDirs | QDir::NoDotAndDotDot);
+
+    // Add folders under /media/$USER to devices.
+    QDir mediaUserDir(QStringLiteral("/media/") + QString::fromLocal8Bit(qgetenv("USER")));
+    databaseLocations += mediaUserDir.entryInfoList(
+            QDir::AllDirs | QDir::NoDotAndDotDot);
+
+    // Add folders under /run/media/$USER to devices.
+    QDir runMediaUserDir(QStringLiteral("/run/media/") + QString::fromLocal8Bit(qgetenv("USER")));
+    databaseLocations += runMediaUserDir.entryInfoList(
+            QDir::AllDirs | QDir::NoDotAndDotDot);
+#elif defined(__APPLE__)
+    databaseLocations.append(QDir(QStringLiteral("/Volumes")).entryInfoList(QDir::AllDirs | QDir::NoDotAndDotDot));
+#endif
+
+    foreach (QFileInfo databaseLocation, databaseLocations) {
+        QDir databaseDir = QDir(databaseLocation.filePath());
+        if (!databaseDir.cd(kDatabaseDirectory)) {
+            continue;
+        }
+
+        if (!databaseDir.exists(kDatabaseFilename)) {
+            continue;
+        }
+
+        TreeItem* foundDatabase = new TreeItem(seratoFeature);
+
+        QString displayPath = databaseLocation.filePath();
+        if (displayPath.endsWith("/")) {
+            displayPath.chop(1);
+        }
+
+        foundDatabase->setLabel(displayPath);
+
+        QList<QVariant> data;
+        data << QVariant(databaseDir.path())
+             << QVariant(false);
+        foundDatabase->setData(data);
+
+        foundDatabases << foundDatabase;
+    }
+
+    return foundDatabases;
+}
+
+void clearTable(QSqlDatabase& database, QString tableName) {
+    QSqlQuery query(database);
+    query.prepare("DELETE FROM " + tableName);
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query) << "tableName:" << tableName;
+        return;
+    }
+
+    query.prepare("DELETE FROM sqlite_sequence WHERE name=:name");
+    query.bindValue(":name", tableName);
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query) << "tableName:" << tableName;
+        return;
+    }
+
+    qDebug() << "Serato table entries of '" << tableName << "' have been cleared.";
+}
+
+} // anonymous namespace
+
+SeratoPlaylistModel::SeratoPlaylistModel(QObject* parent,
+        TrackCollectionManager* trackCollectionManager,
+        QSharedPointer<BaseTrackCache> trackSource)
+        : BaseExternalPlaylistModel(parent, trackCollectionManager, "mixxx.db.model.serato.playlistmodel", "serato_playlists", "serato_playlist_tracks", trackSource) {
```

Comment:
Please break long lines around column 80.

Reply:
I did, but clang-format unwrapped them again. We need to fix that in the config.

Reply:
It will not, if you put every argument on a single line.